### PR TITLE
Unit test app synchronization

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
This PR will add app synchonization tests
Fixes issue with the app update

### CODE REVIEW

**Comment by mickmister** [comment]
@hanzei Why is the `Awaiting Submitter Action` label on the PR?

File: server/proxy/synchronize_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+			for _, app := range installedApps {
+				if _, isListed := listedApps[app.AppID]; !isListed {
+					require.False(t, p.AppIsEnabled(app))
+				}
+			}
+		})
+	}
+}
+
+func createProxy(a map[apps.AppID]*apps.App, m map[apps.AppID]*apps.Manifest) *Proxy {
```
Comment (mickmister) [inline server/proxy/synchronize_test.go:184]: Can we name this `createTestProxy`, so it's obvious to callers that it's creating a `Proxy` for testing?

**Review by mickmister — APPROVED** [review]
LGTM :+1:

**Comment by hanzei** [comment]
@mickmister Merge conflicts

**Comment by iomodo** [comment]
@levb kind reminder to review this. Besides tests it has a bug fix: when calling an `onVersionChanged` we are not supplying the `appID` to the request

**Comment by levb** [comment]
I'd like to close this PR. The approach taken is rather heavy for the value, and it'll take a lot of work to sync up with `master` at this point. @catalintomai @mickmister @iomodo any objections?

**Comment by iomodo** [comment]
No objections

**Comment by catalintomai** [comment]
@levb, none either. Might be obsolete anyway.

**Pull request closed without merging**